test/Service/SessionManagerFactoryTest.php

```diff
@@ -118,13 +130,12 @@ public function testFactoryWillAddValidatorViaConfiguration(): void
 
         self::assertInstanceOf(ManagerInterface::class, $manager);
 
-        $chain = $manager->getValidatorChain();
-
-        self::assertInstanceOf(EventManager::class, $chain);
-
-        $listeners = iterator_to_array($this->getListenersForEvent('session.validate', $chain));
-
-        self::assertCount(2, $listeners);
+        $containedValidators = $this->getReflectionProperty($manager, 'validators');
+        self::assertIsArray($containedValidators);
+        self::assertCount(2, $containedValidators);
+        foreach ($containedValidators as $validator) {
+            self::assertTrue(is_a($validator, Validator\ValidatorInterface::class, true));
+        }
```

Comment:
Can we change this test so that:

- Create a manager with non-default validators
- Fix the environment so it will fail validation
- Assert validation fails

Reaching into private properties via reflection is brittle - better to test behaviour than code.